server/routes_test.go

```diff
@@ -82,9 +83,28 @@ func TestRouteAdvertise(t *testing.T) {
 	if srvA.routeInfo.IP != "nats-route://example.com:80/" {
 		t.Fatalf("Expected srvA.routeInfo.IP to be set, got %v", srvA.routeInfo.IP)
 	}
+	srvA.mu.Unlock()
+	srvSeed.mu.Lock()
 	if srvSeed.routeInfo.IP != "" {
 		t.Fatalf("Expected srvSeed.routeInfo.IP to not be set, got %v", srvSeed.routeInfo.IP)
 	}
+	srvSeed.mu.Unlock()
+
+	// create a TCP client, connect to srvA Cluster and verify info.
+	testCn, _ := net.Dial("tcp", net.JoinHostPort(optsA.Cluster.Host, strconv.Itoa(optsA.Cluster.Port)))
```

Comment:
This is not what I had in mind. I would want to see a server connecting to the seed and receive a second INFO with the details about srvA. Probably this test could be moved to `test/` so you can have access to `createRouteConn()`, etc... see `TestSeedReturnIPInInfo` for instance.